packages/perseus-editor/src/components/locked-figures-section.tsx

```diff
@@ -0,0 +1,87 @@
+/**
+ * LockedFiguresSection is a section of the InteractiveGraphEditor that allows
+ * the user to add and remove locked figures from the graph. It includes
+ * the dropdown for adding figures as well as the settings for each figure.
+ */
+import {View, useUniqueIdWithMock} from "@khanacademy/wonder-blocks-core";
+import {spacing} from "@khanacademy/wonder-blocks-tokens";
+import {StyleSheet} from "aphrodite";
+import * as React from "react";
+
+import LockedFigureSelect from "./locked-figure-select";
+import LockedFigureSettings from "./locked-figure-settings";
+import {getDefaultFigureForFigureType} from "./util";
+
+import type {Props as InteractiveGraphEditorProps} from "../widgets/interactive-graph-editor";
+import type {LockedFigure, LockedFigureType} from "@khanacademy/perseus";
+
+type Props = {
+    figures?: Array<LockedFigure>;
+    onChange: (props: Partial<InteractiveGraphEditorProps>) => void;
+};
+
+const LockedFiguresSection = (props: Props) => {
+    const uniqueId = useUniqueIdWithMock().get("locked-figures-section");
+    const {figures, onChange} = props;
+
+    function addLockedFigure(newFigure: LockedFigureType) {
+        const lockedFigures = figures || [];
+        const newProps = {
+            lockedFigures: [
+                ...lockedFigures,
+                getDefaultFigureForFigureType(newFigure),
+            ],
+        };
+        onChange(newProps);
+    }
+
+    function removeLockedFigure(index: number) {
+        const lockedFigures = figures || [];
+        onChange({
+            lockedFigures: [
+                ...lockedFigures.slice(0, index),
+                ...lockedFigures.slice(index + 1),
+            ],
+        });
+    }
+
+    function changeCoord(index: number, coord: [number, number]) {
+        const lockedFigures = figures || [];
+        const newProps = {
+            lockedFigures: [
+                ...lockedFigures.slice(0, index),
+                {
+                    ...lockedFigures[index],
+                    coord,
+                },
+                ...lockedFigures.slice(index + 1),
+            ],
+        };
+        onChange(newProps);
+    }
+
+    return (
+        <View style={styles.container}>
+            {figures?.map((figure, index) => (
+                <LockedFigureSettings
+                    key={`${uniqueId}-locked-${figure}-${index}`}
+                    {...figure}
+                    onRemove={() => removeLockedFigure(index)}
```

Comment:
I have a possibly-unfounded concern that using indexes for these callbacks could be problematic.  I know for react, you aren't supposed to use array indexes for keys.  Does that apply here?